File: 0001-two-sum/0001-two-sum.java
import java.util.HashMap;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> dict = new HashMap<>();
        int[] result = new int[2];
        for (int i = 0; i < nums.length ; i++) {
            int complement = target - nums[i];

            if (dict.containsKey(complement)) {
                
                result[0] = dict.get(complement);
                result[1] = i;
            }

            dict.put(nums[i], i);

        }
        return result;

    }
}

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        values = {}

        for index, number in enumerate(nums):
            difference = target - number

            if difference in values:
                return [values[difference], index]
            else:
                values[number] = index
            
        return []
            


        
        